src/streamlink/stream/dash/manifest.py

```diff
@@ -768,6 +771,16 @@ def segments(
                 byterange=segment_url.media_range,
             )
 
+    def calculate_optimal_start(self) -> int:
+        """Calculate the optimal segment number to start based on the suggestedPresentationDelay"""
+        suggested_delay = self.root.suggestedPresentationDelay
+
+        offset = max(0, math.ceil(suggested_delay.total_seconds() / self.duration_seconds))
```

Comment:
`self.duration_seconds` can be `0.0` if the optional `duration` attribute is missing (unknown segment duration). This probably will never be the case for dynamic manifests, but we still have to cover this case, or we raise a `ZeroDivisionError`.

In this case, the method should choose a constant number that's referenced from a class attribute (see `MPD.DEFAULT_MINBUFFERTIME`). Not sure though what would be a good value, maybe `3` (default value of `hls-live-edge`). In the future, we can make this value, as well as the live-edge offset calculation customizable using a new session option.

My idea is deprecating and renaming the `hls-live-edge` option. Then we can change it so time values can be set too with a time-unit suffix, instead of just whole segment numbers.

Reply:
I've added this check, and also added a test for this scenario.